specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json

```diff
@@ -0,0 +1,291 @@
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create DDoS custom policy": {
+            "$ref": "./examples/DdosCustomPolicyCreate.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DdosCustomPolicy": {
+      "description": "A DDoS custom policy in a resource group.",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },        
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DdosCustomPolicyPropertiesFormat",
+          "description": "Properties of the DDoS custom policy."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      }
+    },
+    "DdosCustomPolicyPropertiesFormat": {
+      "properties": {
+        "resourceGuid": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource GUID property of the DDoS custom policy resource. It uniquely identifies the resource, even if the user changes its name or migrate the resource across subscriptions or resource groups."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state of the DDoS custom policy resource. Possible values are: 'Succeeded', 'Updating', 'Deleting', and 'Failed'."
+        },
+        "publicIPAddresses": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "id":"string"
+            }
+          },
+          "description": "The list of public IPs associated with the DDoS custom policy resource. This list is read-only."
+        },
+        "protocolCustomSettings": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./network.json#/definitions/SubResource" 
+          },
+          "description": "The list of public IPs associated with the DDoS custom policy resource. This list is read-only."
+        }
+      },
+      "description": "DDoS custom policy properties."
+    },
+    "ProtocolCustomSettingsFormat": {
+      "properties": {
+        "protocol": {
+          "readOnly": false,
```

Comment:
Add enum for protocol please? Tcp, Udp and Syn

Reply:
done